themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/applications-page/ApplicationsPage.tsx

```diff
@@ -15,21 +15,196 @@
+          {this.state.applications.map((application: Application, appIndex: number) => {
+            return (
+              <DataListItem key={'application-' + appIndex} aria-labelledby="applications-list" isExpanded={this.state.isRowOpen[appIndex]}>
+                <DataListItemRow>
+                  <DataListToggle
+                    onClick={() => this.onToggle(appIndex)}
+                    isExpanded={this.state.isRowOpen[appIndex]}
+                    id={'applicationToggle' + appIndex}
+                    aria-controls="expandable"
+                  />
+                  <DataListItemCells
+                    dataListCells={[
+                      <DataListCell width={2} key={'app-' + appIndex}>
+                        <BuilderImageIcon /> <Text component={TextVariants.small}>{application.clientId}</Text>
+                      </DataListCell>,
+                      <DataListCell width={2} key={'internal-' + appIndex}>
+                        <span>{application.inUse && !application.userConsentRequired ? 'Internal' : 'Third-party'}
+                          {application.offlineAccess ? ', Offline Access' : ''}
+                        </span>
+                      </DataListCell>,
+                      <DataListCell width={2} key={'status-' + appIndex}>
+                        <span>{application.inUse ? 'In use' : 'Not in use'}</span>
+                      </DataListCell>,
+                      <DataListCell width={4} key={'baseUrl-' + appIndex}>
+                        <a href={application.baseUrl}><LinkIcon /> {application.baseUrl}</a>
+                      </DataListCell>,
+                    ]}
+                  />
+                </DataListItemRow>
+                <DataListContent
+                  noPadding={false}
+                  aria-label="Application Details"
+                  id="expandable"
+                  isHidden={!this.state.isRowOpen[appIndex]}
+                >
+                  <Grid sm={12} md={12} lg={12}>
+                    <GridItem><b>{'Client: '}</b> {application.clientId}</GridItem>
+                    {application.description &&
+                      <GridItem><b>{'Description: '}</b> {application.description}</GridItem>
+                    }
+                    <GridItem><b>{'URL: '}</b> {application.baseUrl}</GridItem>
+                    {application.consent &&
+                      <React.Fragment>
+                        <GridItem span={12}>
+                          <b>Has access to:</b>
+                        </GridItem>
+                        {application.consent.grantedScopes.map((scope: GrantedScope, scopeIndex: number) => {
+                          return (
+                            <React.Fragment key={'scope-' + scopeIndex} >
+                              <GridItem offset={1}><CheckIcon /> {scope.name}</GridItem>
+                            </React.Fragment>
+                          )
+                        })}
```

Comment:
I'm not sure what but IMHO it's missing something. :) (I'm definitely no UX expert but my common sense tells me it could _somehow_ look better. :)). Perhaps the list has too big left margin(?) I don't know ;)

Maybe a simple one-line list ("Has access to: email, profile, roles") would look better?

![Screenshot 2019-11-29 at 16 36 44](https://user-images.githubusercontent.com/13906855/69878944-7e4cac00-12c6-11ea-89fb-f46b3af42474.png)

Reply:
@vmuzikar funny enough, it was a bit tricky to display things this way and what you suggested is super easy :) I tried to keep things in sync with the wireframes, but maybe we should change.

@ssilvert what do you think?

Reply:
Not sure without seeing it.  Maybe sketch some ideas and go with what you think looks best?

Reply:
@ssilvert in summary instead of display "Has access to:" which check marks, that would be displayed in a single line. Something like:

**Has Access to:** email, profile, roles

Reply:
@vmuzikar based on the discussion from the mailing list, I will mark this as resolved. Let's stick with the wireframes and for future improvements we can discuss with UXD team.

I will revisit all the details from the wireframe and update this PR.